src/backend/joanie/payment/models.py

```diff
@@ -1 +1,430 @@
+        db_table = "joanie_invoice"
+        verbose_name = _("Invoice")
+        verbose_name_plural = _("Invoices")
+
+    def __str__(self):
+        return f"Invoice {self.reference}"
+
+    def _get_context_for_locale(self, language_code=None):
+        """
+        Generate invoice context.
+        """
+        language_code = language_code or get_language()
+        with switch_language(self.order.product, language_code):
+            vat = D(settings.JOANIE_VAT)
+            total_amount = self.amount
+            vat_amount = total_amount * vat / 100
+            net_amount = total_amount - vat_amount
+            currency = settings.JOANIE_CURRENCY[1]
+
+            return {
+                "metadata": {
+                    "issued_on": str(self.created_on),
+                    "reference": self.reference,
+                    "type": "invoice",
+                },
+                "order": {
+                    "amount": {
+                        "currency": currency,
+                        "subtotal": str(net_amount),
+                        "total": str(total_amount),
+                        "vat_amount": str(vat_amount),
+                        "vat": str(vat),
+                    },
+                    "company": settings.JOANIE_INVOICE_COMPANY_CONTEXT,
+                    "customer": {
+                        "address": self.recipient_address,
+                        "name": self.recipient_name,
+                    },
+                    "product": {
+                        # pylint: disable=no-member
+                        "description": self.order.product.description,
+                        # pylint: disable=no-member
+                        "name": self.order.product.title,
+                    },
+                    "seller": {
+                        "address": settings.JOANIE_INVOICE_SELLER_ADDRESS,
+                    },
+                },
+            }
+
+    def _create_localized_contexts(self):
+        """
+        Update or create the invoice context for all languages.
+
+        In order to save space, we want to generate related document only on
+        demand. Furthermore, we need to store product description to prevent
+        inconsistency in case of product would be updated. That's why we create
+        a translated context for all active language, then we are able to
+        generate invoice document with consistent data only on demand.
+        """
+        for language, __ in settings.LANGUAGES:
+            localized_context = self._get_context_for_locale(language)
+            # pylint: disable=no-member
+            self.translations.create(language_code=language, context=localized_context)
+
+    @property
+    def document(self):
+        """
+        Get the document related to the invoice instance;
+        """
+        # pylint: disable=no-member
+        document = InvoiceDocument(context_query=self.context)
+        return document.create(persist=False)
```

Comment:
The property name and the docstring suggests that we can use it as a "get" but the create would then fail?

Reply:
It seems to be a naming issue in Marion. The method should be renamed to "generate" in marion? ping @jmaupetit

Reply:
Yes, this is not something that is blocking. Good idea to open an issue in Marion.